### PR TITLE
Introduce response status errors

### PR DESCRIPTION
**Pull request by cthulhu-rider**
Related #150.

Docs will be added right after structural finalization.

### CODE REVIEW

**Review by alexvanin — APPROVED** [review]
I think it is time to merge it finally.

**Review by carpawell — APPROVED** [review]

File: accounting/service.proto [hunk 1]
```diff
@@ -16,6 +16,11 @@ import "session/types.proto";
 // accounts are possible, if both use the same token type.
 service AccountingService {
   // Returns the amount of funds in GAS token for the requested NeoFS account.
+  //
+  // Statuses:
+  // - **OK** (0, [SECTION_SUCCESS](status.md#neo.fs.v2.status.Success)):
```
Comment (realloc) [inline accounting/service.proto:21]: GitHub markdown cross referencing must not be used in API spec.

**Review by realloc — CHANGES_REQUESTED** [review]

File: object/service.proto [hunk 2]
```diff
@@ -26,32 +31,62 @@ service ObjectService {
...
+  // * [Common failures](status.md#neo.fs.v2.status.CommonFail).
   rpc Head(HeadRequest) returns (HeadResponse);
 
   // Search objects in container. Search query allows to match by Object
   // Header's filed values. Please see the corresponding NeoFS Technical
   // Specification section for more details.
+  //
+  // Statuses:
+  // * **OK** (0, [SECTION_SUCCESS](status.md#neo.fs.v2.status.Success)):
+  // objects was selected successfully;
```
Comment (EliChin) [inline object/service.proto:66]: objects were/ object(s) was(were)

File: proto-docs/container.md [hunk 3]
```diff
@@ -94,20 +99,35 @@ response immediately. After a new block is issued in sidechain, request is
...
 | Name | Input | Output |
 | ---- | ----- | ------ |
 | Delete | [DeleteRequest](#neo.fs.v2.container.DeleteRequest) | [DeleteResponse](#neo.fs.v2.container.DeleteResponse) |
 #### Method Get
 
 Returns container structure from `Container` smart contract storage.
 
+Statuses:
+- **OK** (0, [SECTION_SUCCESS](status.md#neo.fs.v2.status.Success)):
+container read successfully;
```
Comment (EliChin) [inline proto-docs/container.md:116]: 'was' is omitted here and in lines 128, 155. It's ok but in all other lines passive voice is used in its full form, so I'd love to see it uniform.

File: proto-docs/object.md [hunk 4]
```diff
@@ -130,6 +150,11 @@ Search objects in container. Search query allows to match by Object
 Header's filed values. Please see the corresponding NeoFS Technical
 Specification section for more details.
 
+Statuses:
+* **OK** (0, [SECTION_SUCCESS](status.md#neo.fs.v2.status.Success)):
+objects was selected successfully;
```
Comment (EliChin) [inline proto-docs/object.md:155]: https://github.com/nspcc-dev/neofs-api/pull/176/files#r749160360

File: proto-docs/status.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+
+<a name="neo.fs.v2.status.Status"></a>
+
+### Message Status
+Declares the general format of the status returns of the NeoFS RPC protocol.
+Status is present in all response messages. Each RPC of NeoFS protocol
+describes the possible outcomes and details of the operation.
+
+Enumerated with code. Any unique result of an operation in NeoFS is
```
Comment (EliChin) [inline proto-docs/status.md:33]: What is enumerated with code? I can guess but it doesn't sound clear enough.

File: proto-docs/status.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+  `G = 1024 * S + L`.
+
+All outcomes are divided into successful and failed, which corresponds
+to the success or failure of the operation. The definition of success
+follows from the semantics of RPC and the description of its purpose.
+The server must not attach code that is the opposite of outcome type.
+
+See the set of return codes in the description for calls.
+
+Each status can carry developer-facing error message. It should human
```
Comment (EliChin) [inline proto-docs/status.md:53]: It should be human...

File: proto-docs/status.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+readable text in English. The server should not transmit (and the client
+should not expect) useful information in the message. Field `details`
+should be used to detail the return.
```
Comment (EliChin) [inline proto-docs/status.md:56]: ...should make the return more detailed.

File: proto-docs/status.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+| ----- | ---- | ----- | ----------- |
+| code | [uint32](#uint32) |  | The status code. |
+| message | [string](#string) |  | Developer-facing error message. |
+| details | [Status.Detail](#neo.fs.v2.status.Status.Detail) | repeated | Data detailing the outcome of the operation. Must be unique by ID. |
+
+
+<a name="neo.fs.v2.status.Status.Detail"></a>
+
+### Message Status.Detail
+Return detail. Contains additional information that can be used to analyze
```
Comment (EliChin) [inline proto-docs/status.md:69]: It contains...

File: accounting/service.proto (hunk 1 above)
Comment (cthulhu-rider) [reply accounting/service.proto:21]: Removed.

File: object/service.proto (hunk 2 above)
Comment (cthulhu-rider) [reply object/service.proto:66]: Fixed.

File: proto-docs/container.md (hunk 3 above)
Comment (cthulhu-rider) [reply proto-docs/container.md:116]: Made the same

File: proto-docs/object.md (hunk 4 above)
Comment (cthulhu-rider) [reply proto-docs/object.md:155]: Auto-generated, so auto-fixed

File: proto-docs/status.md (hunk 5 above)
Comment (cthulhu-rider) [reply proto-docs/status.md:33]: Paraphrased.

File: proto-docs/status.md (hunk 6 above)
Comment (cthulhu-rider) [reply proto-docs/status.md:53]: Fixed.

File: proto-docs/status.md (hunk 7 above)
Comment (cthulhu-rider) [reply proto-docs/status.md:56]: Changed.

File: proto-docs/status.md (hunk 8 above)
Comment (cthulhu-rider) [reply proto-docs/status.md:69]: Fixed.

**Review by EliChin — APPROVED** [review]

File: status/types.proto [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+// The server must not attach code that is the opposite of outcome type.
+//
+// See the set of return codes in the description for calls.
+//
+// Each status can carry developer-facing error message. It should be human
+// readable text in English. The server should not transmit (and the client
+// should not expect) useful information in the message. Field `details`
+// should make the return more detailed.
+message Status {
+    // The status code.
```
Comment (alexvanin) [inline status/types.proto:36]: No periods there and down below if comment is one sentence long.

File: status/types.proto [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+enum Success {
+    // [**0**] Default success. Not detailed.
+    // If the server cannot match successful outcome to the code, it should
+    // use this code.
+    OK = 0;
+}
+
+// Section of failed statuses independent of the operation.
+enum CommonFail {
+    // [**1024**] Internal server error, default failure. Not detailed..
```
Comment (alexvanin) [inline status/types.proto:79]:
```suggestion
    // [**1024**] Internal server error, default failure. Not detailed.
```

**Review by alexvanin — APPROVED** [review]

File: status/types.proto [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+    // If the server cannot match failed outcome to the code , it should
```
Comment (carpawell) [inline status/types.proto:80]:
```suggestion
    // If the server cannot match failed outcome to the code, it should
```

**Review by carpawell — APPROVED** [review]

**Comment by carpawell** [comment]
Can you, please, clarify to me (not in code) how to interpret such common comments:
```
// Statuses:
  // - **OK** (0, SECTION_SUCCESS):
  // balance has been successfully read;
  // - Common failures (SECTION_FAILURE_COMMON).
```

File: session/types.proto [hunk 12]
```diff
@@ -184,6 +185,9 @@ message ResponseMetaHeader {
 
   // `ResponseMetaHeader` of the origin request
   ResponseMetaHeader origin = 5 [json_name = "origin"];
+
+  // Status return.
```
[the remark this answers is not in the document]
Comment (cthulhu-rider) [reply session/types.proto:189]: Removed.

File: status/types.proto (hunk 9 above)
Comment (cthulhu-rider) [reply status/types.proto:36]: Fixed.

File: status/types.proto (hunk 10 above)
Comment (cthulhu-rider) [reply status/types.proto:79]: Fixed

File: status/types.proto (hunk 11 above)
Comment (cthulhu-rider) [reply status/types.proto:80]: Fixed.

**Comment by cthulhu-rider** [comment]
@carpawell, common scheme is:
```
- <code_name> (<global_code>, <section_name>): <desc>
```
Сommon failures are special: they apply equally to all operations. It is inconvenient to copy them in manual mode, and there is no autogeneration, at least not yet. Therefore, the description of the codes of this section should be found in the place of their definition.

**Pull request merged**